docs/src/views/ShorthandProps.tsx

```diff
@@ -1,37 +1,263 @@
+        </p>
+
+        {codeExample([
+          `<>`,
+          `  <Button icon='chess rook' />`,
+          ``,
+          `  {/* has identical effect to the previous one */}`,
+          `  <Button icon={{ name: 'chess rook' }} />`,
+          `</>`,
+        ])}
+        <p>
+          This works because {code('name')} is the default prop of slot's {code('<Icon />')}{' '}
+          element.
+        </p>
+
+        {/* SHORTHAND AS REACT ELEMENT */}
+        {header('Shorthand value as React Element')}
+        <p>
+          There are cases where it might be necessary to customize element tree that will be
+          rendered as a slot's value. Returning to {code('Button')} example, we might want to render{' '}
+          {code('<i />')} instead of default {code('<Icon />')} element. In that case necessary
+          element might be directly provided as shorthand value:
+        </p>
+        {codeExample([`<Button icon={<i class='my-icon'></i>} />`])}
+
+        <blockquote>
+          <strong>
+            There is a very important caveat here, though: whenever React Element is directly used
+            as a shorthand value, it becomes client's responsibility to handle some aspects that
+            Stardust was originally responsible for (such as {code('styles')} or{' '}
+            {code('accessibility')}).
+          </strong>{' '}
+          This is because, in contrast to other forms of shorthand values, Stardust-evaluated props
+          cannot be unsafely passed to the element which type is, generally, doesn't allow Stardust
+          to make any prior assumptions about. Due to this limitation, you should strive to use
+          other options for shorthand values whenever is possible - for instance, this is how
+          previous example can be rewritten:
+        </blockquote>
+        {codeExample([`<Button icon={{ as: 'i', className: 'my-icon' }} />`])}
+
+        <p>
+          However, there still might be cases where it would be impossible to use object form of the
+          shorthand value - for example, you might want to render some custom elements tree for the
+          slot, while wanting to preserve calculated styles and accessibility aspects for it. In
+          that case {code('render')} callback of function shorthand value should be used.
+        </p>
+
+        {/* SHORTHAND AS A FUNCTION */}
+        {header('Shorthand value as Function')}
+        <p>
+          Providing function as a shorthand value is the most involving but, at the same time, the
+          most powerful option for customizing component's slot. The only requirements for this
+          function are:
+        </p>
+
+        <ul>
+          <li>it should finish syncronously</li>
+          <li>it should return React Element as a result</li>
+        </ul>
+
+        <p>Thus, in its simplest form, it could be used the following way:</p>
+
+        {codeExample([`<Button icon={() => <Icon name='chess rook' />} />`])}
```

Comment:
Not sure that this example brings any value... Let's try not to add examples which should never be used.

Reply:
this one provides a visibility of what does it mean to satisfy criterias of this function. If we will start right with the render callback stuff, it won't be absolutely clear what is going on - there might be a false assumption from the client that this 'render' function is, actually, triggers rendering somewhere, while, in fact, it just returns an element.

I would really like to leave this simple explanation of what does it mean to pass function as a shorthand value, and what Stardust expects from this function.

What do you think?